src/govuk/i18n.mjs

```diff
@@ -15,19 +15,56 @@ function I18n (translations) {
  * returns the appropriate string.
  *
  * @param    {String}  lookupKey  - The lookup key of the string to use.
+ * @param    {Object}  options    - Any options passed with the translation string, e.g: for string interpolation.
  * @returns  {String}             - The appropriate translation string.
  */
-I18n.prototype.t = function (lookupKey) {
+I18n.prototype.t = function (lookupKey, options) {
   if (!lookupKey) {
     // Print a console error if no lookup key has been provided
     throw new Error('i18n lookup key missing.')
   } else if (lookupKey in this.translations) {
-    // Return the translation string if it exists in the object
-    return this.translations[lookupKey]
+    // Fetch the translation string for that lookup key
+    var translationString = this.translations[lookupKey]
+
+    // Check for ${} placeholders in the translation string
+    if (translationString.match(/\${(.\S+)}/)) {
+      if (!options) {
+        throw new Error('i18n: cannot replace placeholders in string if no option data provided')
+      }
```

Comment:
Is this needed if we're throwing errors when the placeholders aren't found?

Reply:
You're right, the error will be flagged later on. However, it catches the error slightly earlier, so it doesn't have to do the regex lookup and while loop before it throws.